pkg/nodeidentity/openstack/identify.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openstack
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/kops/pkg/nodeidentity"
+)
+
+// MetadataLatest url for latest metadata which contains instance metadata
+const MetadataLatest string = "http://169.254.169.254/openstack/latest/meta_data.json"
+
+// nodeIdentifier identifies a node
+type nodeIdentifier struct {
+}
+
+// New creates and returns a nodeidentity.Identifier for Nodes running on OpenStack
+func New() (nodeidentity.Identifier, error) {
+	return &nodeIdentifier{}, nil
+}
+
+// IdentifyNode queries OpenStack for the node identity information
+func (i *nodeIdentifier) IdentifyNode(ctx context.Context, node *corev1.Node) (*nodeidentity.Info, error) {
+	metadata, err := getLocalMetadata()
+	if err != nil {
+		return nil, err
+	}
+
+	info := &nodeidentity.Info{}
+	info.InstanceGroup = metadata.UserMeta.InstanceGroup
+
+	return info, nil
+}
+
+func getLocalMetadata() (*InstanceMetadata, error) {
+	var meta InstanceMetadata
+	var client http.Client
+	resp, err := client.Get(MetadataLatest)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusOK {
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, err
+		}
+		err = json.Unmarshal(bodyBytes, &meta)
+		if err != nil {
+			return nil, err
+		}
+		return &meta, nil
+	}
+	return nil, err
```

Comment:
This `err` variable will always be `nil` here, maybe return an `errors.New()` or `fmt.Errorf()` instead?